Lib/enum.py

```diff
@@ -564,30 +595,67 @@ def __new__(cls, value):
             pass
         except TypeError:
             # not there, now do long search -- O(n) behavior
-            for member in cls._member_map_.values():
-                if member._value_ == value:
+            for member in cls._unique_member_map_.values():
+                if member.value == value:
                     return member
         # still not found -- try _missing_ hook
+
+        # TODO: Maybe remove try/except block and setting __context__ in this case?
         try:
-            exc = None
             result = cls._missing_(value)
         except Exception as e:
-            exc = e
-            result = None
+            # Huge boost for standard enum
+            if cls._missing_ is Enum._missing_:
+                raise
+            else:
+                e.__context__ = ValueError(f'{value!r} is not a valid {cls.__qualname__}')
+                raise
```

Comment:
With this try/except block code runs twice slower for non-enum values checks, then without it . 
Do we really need to bother checking all errors and add `ValueError` to context , knowing it will happen either in standard `Enum._missing_` or in user-defined code? In fact, `Flag._missing_` also raises same error as `Enum._missing_` and handled and rerised too.